File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_cur = float(inf)
        max_profit = 0
        for price in prices:
            max_profit = max(max_profit, price - min_cur)
            min_cur = min(min_cur, price)

        return max_profit

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min_price = Integer.MAX_VALUE;
        int max_profit = 0;
        for (int price : prices) {
            max_profit = Math.max(max_profit, price - min_price);
            min_price = Math.min(min_price, price);
        }
        return max_profit;
    }
}